### PR TITLE
[FLINK-21641][python] Support 'useModules' and 'listFullModules' in PyFlink

### PR DESCRIPTION
**Pull request by LadyForest**
## What is the purpose of the change

This pull request supports `useModules` and `listFullModules` in PyFlink to keep Table API consistent.

## Brief change log

  - Add implementation of interface method `useModules` and `listFullModules` in `TableEnvironment`.
  - Add class `ModuleEntry` for converting result of `useModules`.
  - Add class `ModuleMock` for testing.

## Verifying this change

This change added tests and can be verified as follows:

 - Remove `loadModule/unloadModule/listModules/useModules/listFullModules` from `EnvironmentAPICompletenessTests`.
 - Add more tests for `load_modules` and `unload_modules` in `TableEnvironmentTest` and `BatchTableEnvironmentTests`.
 - Add tests for `use_modules` and `list_full_modules` in `TableEnvironmentTest` and `BatchTableEnvironmentTests`.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (**yes** / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / **not documented**)
     - Docs will be updated after FLINK-21300 finish.

### CODE REVIEW

**Review by dianfu — COMMENTED** [review]
@LadyForest Thanks for the PR! LGTM overall. It would be great to also update the documentation: https://ci.apache.org/projects/flink/flink-docs-master/docs/dev/table/modules/

File: flink-python/pyflink/table/module_entry.py [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+from pyflink.java_gateway import get_gateway
+
+__all__ = ['ModuleEntry']
+
+
+class ModuleEntry(object):
```
Comment (dianfu) [inline flink-python/pyflink/table/module_entry.py:23]: What about moving this class to module.py?

**Comment by LadyForest** [comment]
> @LadyForest Thanks for the PR! LGTM overall. It would be great to also update the documentation: https://ci.apache.org/projects/flink/flink-docs-master/docs/dev/table/modules/

Hi, @dianfu thanks for the comments. There is a separate Jira task FLINK-21300 to track docs update. It's a subtask of FLINK-21045.

File: flink-python/pyflink/table/module_entry.py (hunk 1 above)
Comment (LadyForest) [reply flink-python/pyflink/table/module_entry.py:23]: > What about moving this class to module.py?

+1.

File: flink-python/src/main/java/org/apache/flink/python/util/ModuleMock.java [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.python.util;
+
+import org.apache.flink.table.module.Module;
+
+/** Mocking {@link Module} for Python tests. */
+public class ModuleMock implements Module {
```
Comment (dianfu) [inline flink-python/src/main/java/org/apache/flink/python/util/ModuleMock.java:24]: Could we reuse the ModuleMock defined in flink-table-api-java module? Could use _load_specific_flink_module_jars on how to load classes defined in test jar.

**Comment by LadyForest** [comment]
Hi @dianfu, I've updated the PR, and you can take a look after CI passes.

**Review by dianfu — APPROVED** [review]
@LadyForest Thanks a lot for the update!  LGTM. Will merge once the tests passed.

**Pull request merged**